### PR TITLE
many: generalize wording of NFS workaround

### PR DESCRIPTION
**Pull request by zyga**
Ahead of [introduction of CIFS workaround](https://github.com/snapcore/snapd/pull/13694), generalize the names so that we use more general language rather instead of focusing on NFS.

As a special exception the externally visible wording related to NFS is kept intact in two places:
 - The apparmor "nfs-support" file name
 - The system key "nfs-home" key.

From points of view this is all an elaborate internal rename that should not be
observable outside of snapd, apart from log messages that may no longer speak of
NFS but of remote file systems.

### CODE REVIEW

File: osutil/nfs_linux_test.go [hunk 1]
```diff
@@ -62,37 +62,39 @@ func (s *nfsSuite) TestIsHomeUsingNFS(c *C) {
...
+		isRemoteFS: true,
 	}}
 	for _, tc := range cases {
 		restore := osutil.MockMountInfo(tc.mountinfo)
 		defer restore()
 		restore = osutil.MockEtcFstab(tc.fstab)
 		defer restore()
 
-		nfs, err := osutil.IsHomeUsingNFS()
+		comment := Commentf("mountinfo %#v", tc.mountinfo)
```
Comment (zyga) [inline osutil/nfs_linux_test.go:90]: This is unrelated, I will remove it.

**Review by bboozzoo — APPROVED** [review]
LGTM

File: interfaces/system_key.go [hunk 2]
```diff
@@ -70,6 +70,9 @@ type systemKey struct {
 	// kernel version or similar settings. If those change we may
 	// need to change the generated profiles (e.g. when the user
 	// boots into a more featureful seccomp).
+	//
+	// As an exception, the NFSHome is not renamed to RemoteFSHome
```
[the remark this answers is not in the document]
Comment (zyga) [reply interfaces/system_key.go:74]: heh, yes NFS was the right name all along :)

**Comment by zyga** [comment]
I've rebased this branch, hoping to get tests more green.

File: osutil/nfs_linux.go [hunk 3]
```diff
@@ -26,12 +26,12 @@ import (
 
 var etcFstab = "/etc/fstab"
 
-// isHomeUsingNFS returns true if NFS mounts are defined or mounted under /home.
+// isHomeUsingRemoteFS returns true if NFS mounts are defined or mounted under /home.
```
Comment (ernestl) [inline osutil/nfs_linux.go:29]: "if NFS mounts" in description should probably change?

**Review by ernestl — APPROVED** [review]
Looks good, one comment

**Review by Meulengracht — CHANGES_REQUESTED** [review]
I mean the changes are fine, but what is the point of this? What kinds of remote FSs are relevant of which are not network mounted?

**Comment by zyga** [comment]
@Meulengracht it's so that we don't only talk about "NFS" the file system, not the concept. The follow-up extends this to support CIFS without saying now it's NFS or CIFS everywhere.

**Review by Meulengracht — APPROVED** [review]
Thanks, i'll +1 this, but I still think `NetworkFS/NFS` is better conceptually as a naming, even with the conflict of NFS the file system. I see very few direct mentions that would specifically refer to the NFS file-system, of which could have been reworded.

**Comment by bboozzoo** [comment]
needs a rebase

**Pull request merged**